eclipse-che/che-dashboard
# feat: print the timeout from the CR for the workspace idle/runtimeout…

… warning page

Signed-off-by: dkwon17 <dakwon@redhat.com>



### What does this PR do?
Reads the `spec.devEnvironments.secondsOfInactivityBeforeIdling` and `spec.devEnvironments.secondsOfRunBeforeIdling` for the idle timeout and run timeout warning pages respectively.

<details>
<summary>Workspace inactivity</summary>

<img width="1440" alt="Screen Shot 2022-08-05 at 11 27 15 AM" src="https://user-images.githubusercontent.com/83611742/183110103-a9fe0d8d-495f-456b-985c-faf3c4de9033.png">

</details>

<details>
<summary>Workspace runtime limit </summary>

<img width="1440" alt="Screen Shot 2022-08-05 at 11 33 55 AM" src="https://user-images.githubusercontent.com/83611742/183111297-72e13ba3-0acb-421c-aa1c-7a5e2fc84cac.png">

</details>


### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/21578

### Is it tested? How?


#### Testing the 'Workspace inactivity' error page
1. Set the workspace inactivity timeout to a smaller value by setting checluster.spec.devEnvironments.secondsOfInactivityBeforeIdling to, for example, 120.

2. Start a workspace. Once the editor opens, let it timeout by not interacting with it for about 120 seconds.

3. Once the workspace times out, refresh the page

4. The error page should mention that the idle timeout is 2 minutes.

#### Testing the 'Workspace run timeout' error page
1. Set the workspace run timeout to a value smaller than the inactivity timeout by setting `checluster.spec.devEnvironments.secondsOfRunBeforeIdling` to, for example, `80`.

2. Start a workspace. After the workspace starts, the workspace should terminate in about 80 seconds.

3. Once the workspace times out, refresh the editor page

4. The error page should mention that the run timeout is 1 minute and 20 seconds.

#### Release Notes

The workspace inactivity and workspace run timeout warning pages now display their respective timeout durations.

#### Docs PR
